File: openapi-generator/dart/lib/model/user_tag_role.dart
part of openapi.api;

class UserTagRole {
  /* The name of the role the user has been granted on this tag. */
  String role = null;
  /* The id of the role the user has been granted on this tag. */
  String roleId = null;
  
  UserTagRoleTag tag = null;
  UserTagRole();

  @override
  String toString() {
    return 'UserTagRole[role=$role, roleId=$roleId, tag=$tag, ]';
  }

  UserTagRole.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['role'] == null) {
      role = null;
    } else {
          role = json['role'];
    }
    if (json['roleId'] == null) {
      roleId = null;
    } else {
          roleId = json['roleId'];
    }
    if (json['tag'] == null) {
      tag = null;
    } else {
      tag = new UserTagRoleTag.fromJson(json['tag']);
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'role': role,
      'roleId': roleId,
      'tag': tag
    };
  }

  static List<UserTagRole> listFromJson(List<dynamic> json) {
    return json == null ? new List<UserTagRole>() : json.map((value) => new UserTagRole.fromJson(value)).toList();
  }

  static Map<String, UserTagRole> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, UserTagRole>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new UserTagRole.fromJson(value));
    }
    return map;
  }
}

File: swagger-codegen/csharp-dotnet2/src/main/CsharpDotNet2/samsara/Model/UserTagRole.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace samsara.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class UserTagRole {
    /// <summary>
    /// The name of the role the user has been granted on this tag.
    /// </summary>
    /// <value>The name of the role the user has been granted on this tag.</value>
    [DataMember(Name="role", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "role")]
    public string Role { get; set; }

    /// <summary>
    /// The id of the role the user has been granted on this tag.
    /// </summary>
    /// <value>The id of the role the user has been granted on this tag.</value>
    [DataMember(Name="roleId", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "roleId")]
    public string RoleId { get; set; }

    /// <summary>
    /// Gets or Sets Tag
    /// </summary>
    [DataMember(Name="tag", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "tag")]
    public UserTagRoleTag Tag { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class UserTagRole {\n");
      sb.Append("  Role: ").Append(Role).Append("\n");
      sb.Append("  RoleId: ").Append(RoleId).Append("\n");
      sb.Append("  Tag: ").Append(Tag).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
